Sources/Sentry/SentrySubClassFinder.m

```diff
@@ -100,4 +92,39 @@ - (void)actOnSubclassesOf:(Class)parentClass block:(void (^)(Class))block
     }];
 }
 
+- (int)getClassList:(Class **)classes
+{
+    int numClasses = [self.objcRuntimeWrapper getClassList:NULL bufferCount:0];
+
+    if (numClasses <= 0) {
+        return -1;
+    }
+
+    int memSize = sizeof(Class) * numClasses;
+    *classes = (__unsafe_unretained Class *)malloc(memSize);
+
+    if (classes == NULL && memSize) {
+        [SentryLog logWithMessage:@"Couldn't allocate memory when retrieving class list."
+                         andLevel:kSentryLevelError];
+        return -1;
+    }
+
+    // Don't assign the result getClassList again to numClasses because if a class is registered
+    // in the meantime our buffer would not be big enough and we would crash when iterating over
+    // the classes.
+    int secondNumClasses = [self.objcRuntimeWrapper getClassList:*classes bufferCount:numClasses];
+
+    // When the number of classes changes between the invocation of class_getSuperclass, we run the
+    // risk of accessing bad memory when iterating over all classes, also see
+    // https://github.com/getsentry/sentry-cocoa/issues/1634
```

Comment:
I'm a little skeptical about this solution because I believe `Class` could be deallocated even after the second run of  `getClassList`.

Reply:
Do you mean `classes` could be deallocated? We manage the memory of `classes` so I don't think it will be deallocated. Or do you mean something else?